pkg/services/secrets/kvstore/cache.go

```diff
@@ -0,0 +1,69 @@
+package kvstore
+
+import (
+	"fmt"
+	"sync"
+	"time"
+)
+
+type decryptionCache struct {
```

Comment:
this cache impl is really cool and simple. The one thing I'd like to see added is a sleep loop that clears the cache at some interval. Otherwise there is some memory leak potential.

Reply:
I was wondering if clearing the cache is necessary since we don't usually have thousands of secrets the memory leak would be limited to the amount of secrets we have stored, and performance wise keeping the cache would be better.